cmd/gcp-controller-manager/csr_signer.go

```diff
@@ -64,38 +69,58 @@ func newGKESigner(ctx *controllerContext) (*gkeSigner, error) {
 		ctx:          ctx,
 		retryBackoff: ClusterSigningGKERetryBackoff,
 		validators:   csrValidators(ctx),
+
+		csrs: csrs,
 	}, nil
 }
 
+// handle is called by the generic Kubernetes certificate controller.
 func (s *gkeSigner) handle(csr *capi.CertificateSigningRequest) error {
+	_, _, err := s.handleInternal(csr)
+	return err
+}
+
+func (s *gkeSigner) handleInternal(csr *capi.CertificateSigningRequest) (processed bool, out *capi.CertificateSigningRequest, err error) {
 	if !certificates.IsCertificateRequestApproved(csr) {
-		return nil
+		return false, nil, nil
+	}
+
+	// Ignore CSRs that are not addressed to the default signer.
+	if csr.Spec.SignerName == nil {
```

Comment:
Should we treat nil like legacy signer for backwards compatibility?

Reply:
Yes, that seems like a good idea.